packages/cli/src/databases/migrations/mysqldb/1681134145996-AddUserActivatedProperty.ts

```diff
@@ -0,0 +1,59 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+import { getTablePrefix, logMigrationEnd, logMigrationStart } from '@db/utils/migrationHelpers';
+import { userSettings } from '@/Interfaces';
+
+export class AddUserActivatedProperty1681134145996 implements MigrationInterface {
+	name = 'AddUserActivatedProperty1681134145996';
+
+	async up(queryRunner: QueryRunner): Promise<void> {
+		logMigrationStart(this.name);
+
+		const tablePrefix = getTablePrefix();
+
+		const activatedUsers: userSettings[] = await queryRunner.query(
+			`SELECT DISTINCT sw.userId AS id,
+				JSON_SET(COALESCE(u.settings, '{}'), '$.userActivated', true) AS settings
+			FROM ${tablePrefix}workflow_statistics AS ws
+						JOIN ${tablePrefix}shared_workflow as sw
+							ON ws.workflowId = sw.workflowId
+						JOIN ${tablePrefix}role AS r
+							ON r.id = sw.roleId
+						JOIN ${tablePrefix}user AS u
+							ON u.id = sw.userId
+			WHERE ws.name = 'production_success'
+						AND r.name = 'owner'
+						AND r.scope = 'workflow'`,
+		);
+
+		const updatedUsers = activatedUsers.map((user) =>
+			queryRunner.query(
+				`UPDATE ${tablePrefix}user SET settings = '${JSON.stringify(user.settings)}' WHERE id = '${
+					user.id
+				}' `,
+			),
+		);
+
+		await Promise.all(updatedUsers);
+
+		if (!activatedUsers.length) {
+			await queryRunner.query(
+				`UPDATE ${tablePrefix}user SET settings = JSON_SET(COALESCE(settings, '{}'), '$.userActivated', false)`,
+			);
+		} else {
+			const activatedUserIds = activatedUsers.map((user) => `'${user.id}'`).join(',');
+
+			await queryRunner.query(
+				`UPDATE ${tablePrefix}user SET settings = JSON_SET(COALESCE(settings, '{}'), '$.userActivated', false) WHERE id NOT IN (${activatedUserIds})`,
+			);
+		}
+
+		logMigrationEnd(this.name);
+	}
+
+	async down(queryRunner: QueryRunner): Promise<void> {
+		const tablePrefix = getTablePrefix();
+		await queryRunner.query(
+			`UPDATE ${tablePrefix}user SET settings = JSON_REMOVE(settings, '$.userActivated')`,
+		);
```

Comment:
If the original value for `user.settings` was `NULL`, `JSON_REMOVE()` will not restore it back to `NULL`, is this correct?

Reply:
if the origin value is null JSON_REMOVE returns null and the value is updated to null. So it keeps as it was.